### DISCUSSION TITLE
GeoCAT-comp & GeoCAT-f2py 2023 roadmap

### DISCUSSION THREAD

**Discussion by erogluorhan** [NCAR/geocat-comp#323] [Announcements]
This discussion can serve as: 

1. Announcement of what we are planning to do overall in the 2023 calendar year for GeoCAT-comp & GeoCAT-f2py, as part of the [GeoCAT 2023 roadmap](https://geocat.ucar.edu/pages/roadmap.html)
2. Feedback/input point for the community on our roadmap

**GeoCAT-comp & GeoCAT-f2py 2023 roadmap (from [GeoCAT 2023 roadmap](https://geocat.ucar.edu/pages/roadmap.html)):**

1. Development 
    - New functionality
       - Community feature requests
       - [Existing backlog items](https://github.com/orgs/NCAR/projects/42/views/12)
       - Fortran to Python conversion
    - Improvement, maintenance, and quality of life updates
       - Optimization/benchmarking (SIParCS internship)
       - Support for M1/M2 and Windows platforms
       - Documentation improvements
       - Continue to research and match community packaging standards

2. Open development & community engagement
    - Showcase functionality to community (examples, outreach, cross-posting, talks, etc)

3. Scalability
    - Dask
       - Ensure compatibility & apply best practices (GeoCAT-comp SIParCS internship)
          - Investigate the existing function implementations to check if they were implemented with Dask best practices
          - Ensure it for new implementations
     - Performance analysis
        - Investigate the performance of the existing function implementations with Dask
    - GPU programming
       - Explore

**Comment by anissa111** [comment]
### Iteration 6 Priorities
January 30 - February 26

1. Incoming and currently reported bugs (including M1 support)
2. New functionality
    a. https://github.com/NCAR/geocat-comp/issues/292 - initial development stage
    b. https://github.com/NCAR/geocat-comp/issues/30 - research stage
    c. https://github.com/NCAR/geocat-comp/issues/192 - initial development stage
